### PR TITLE
DR-2193 soft delete without scratch file

### PR DESCRIPTION
**Pull request by tlangs**
Adding the ability to provide a List of UUIDs to the data deletion endpoint. This enables users to simply send a request to TDR for soft-deletes, instead of needing to stage a file with the rowIds to be deleted.

### CODE REVIEW

File: src/main/java/bio/terra/app/controller/DatasetsApiController.java [hunk 1]
```diff
@@ -382,4 +385,31 @@ private void validateAzureIngestParams(IngestRequestModel ingestRequest) {
       throw new InvalidJobParameterException("Invalid ingest parameters detected", errors);
     }
   }
+
+  private void validateDataDeletionRequest(DataDeletionRequest dataDeletionRequest) {
```
Comment (nmalfroy) [inline src/main/java/bio/terra/app/controller/DatasetsApiController.java]: Probably makes sense to have a unit test for this logic

Comment (nmalfroy) [reply src/main/java/bio/terra/app/controller/DatasetsApiController.java]: Also, I this logic should live in a validator class (see `DataDeletionRequestValidator` for example)

File: src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionCopyFilesToBigQueryScratchBucketStep.java [hunk 2]
```diff
@@ -39,6 +43,28 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     DataDeletionRequest dataDeletionRequest = getRequest(context);
     GoogleBucketResource bucketResource =
         FlightUtils.getTyped(workingMap, CommonFlightKeys.SCRATCH_BUCKET_INFO);
+    List<DataDeletionTableModel> tables;
+    if (dataDeletionRequest.getSpecType() == DataDeletionRequest.SpecTypeEnum.GCSFILE) {
+      tables = copyGcsFileSpecPaths(context, dataDeletionRequest, projectId, bucketResource);
+    } else {
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionCopyFilesToBigQueryScratchBucketStep.java]: Should be explicitly checking for `JSONARRAY` and then throw with an unrecognized value

**Review by nmalfroy — CHANGES_REQUESTED** [review]
Looks good! I think the validator thing is the one thing to update

**Comment by samanehsan** [comment]
Since we have the rowIds in the request, could we insert directly into the soft delete table? Or is there a reason we still want to write that to a CSV file and use external tables?

**Comment by tlangs** [comment]
The external table makes sure the rowIds are valid ids that appear in the dataset table. Plus, it allows for bulk-load operations rather than needing to batch API calls. Its also easy to funnel the JSON array into the same code path as GCS files and not need to maintain two distinct operations for soft-delete.

**Review by samanehsan — APPROVED** [review]
Thanks for the explanation! This lgtm ⛄

**Review by nmalfroy — APPROVED** [review]
Awesome!  Thanks for humoring me :). Also, at some point we should bump to junit 5...I think it simplifies some of the data provider stuff

**Pull request merged**